super_dpf.py

```diff
@@ -16,207 +18,216 @@
+            return self._help
         else:
-            aws_settings = self.settings['aws']
-            os.environ['AWS_ACCESS_KEY_ID'] = aws_settings['access_key']
-            os.environ['AWS_SECRET_ACCESS_KEY'] = aws_settings['secret_key']
+            return self.title
 
-            buckets = []
-            s3 = boto3.resource('s3')
-            for s3bucket in s3.buckets.all():
-                if 'dpf' in s3bucket.name:
-                    buckets.append(s3bucket.name)
+    def value_dialog(self, old_value):
+        if old_value:
+            prompt = '{} [{}]:'.format(self.title, old_value)
+            value = raw_input(prompt) or old_value
+        else:
+            value = raw_input('{}: '.format(self.title))
+        if issubclass(self.value_type, basestring):
+            return value
+        elif self.value_type == int:
+            return int(value)
+        else:
+            print('Unsupported value type: {}'.format(self.value_type))
+            return None
 
-            if self.settings['aws']['bucket'] not in buckets:
-                buckets.append(self.settings['aws']['bucket'])
 
-            for bucket in buckets:
-                s3bucket = s3.Bucket(bucket)
-                for s3object in s3bucket.objects.iterator():
-                    filename = '{}/{}'.format(self.aws_dir, s3object.key)
-                    if not os.path.isfile(filename):
-                        logging.info('Downloading {}'.format(s3object.key))
-                        bucket.download_file(s3object.key, filename)
+class BaseDPF(object):
+    REQUIRED_SETTINGS = list()
 
+    def __init__(self, settings_object):
+        self.settings = settings_object
 
-class GPhotoController(BaseDPF):
     @property
-    def gphotos_dir(self):
-        return '{}/{}/'.format(self.PATHS['photos'], 'gphotos')
+    def subdir(self):
+        pk_list = [str(s) for s in self.__class__.settings_pk(self.settings)]
+        return '{}'.format('_'.join(pk_list))
+
+    @classmethod
+    def settings_dialog(cls, old_settings=dict()):
+        settings_object = dict()
+        for item in cls.REQUIRED_SETTINGS:
+            settings_object[item.name] = item.value_dialog(
+                old_settings.get(item.name))
+        return settings_object
+
+    @classmethod
+    def settings_pk(cls, settings_object):
+        pk_values = sorted({i.name for i
+                            in cls.REQUIRED_SETTINGS if i.primary})
+        pk = list()
+        pk.append(cls.__name__)
+        for value in pk_values:
+            pk.append(settings_object.get(value))
+        return pk
+
+    @classmethod
+    def verify_local(cls, cloud_filelist, local_filelist):
+        for local_file in local_filelist:
+            if local_file not in cloud_filelist:
+                logging.info('Deleting local file {} not in {}'.format(
+                    local_file, cls.__name__))
+                os.remove(local_file)
+
+    @staticmethod
+    def stylize_image(image):
```

Comment:
I expect this is more likely to be used at a time when you have an instance of this class instead of just the uninstantiated class.